src/js/parsers/captions/parsers.dfxp.js

```diff
@@ -56,9 +56,10 @@ define([
             utils.tryCatch(function() {
                 var idx = data.indexOf('begin=\"');
                 data = data.substr(idx + 7);
```

Comment:
That is better, but you still need to offset `idxStart`with the length of the attribute name. `idxStart + 7` for `begin="` and `idxStart + 5` for `end="`.

Or better yet, just match the whole thing with a RegExp and forget about all `indexOf`s and `substr`s.

But then again, maybe proper XML-parsing is the best way. The way it is done in Flash-mode in JW Player 6:
https://github.com/jwplayer/jwplayer/blob/master/src/flash/com/longtailvideo/jwplayer/parsers/DFXP.as#L45